src/main/ruby/core/stat.rb

```diff
@@ -56,17 +73,25 @@ class Stat
     S_ISVTX  = Rubinius::Config['rbx.platform.file.S_ISVTX']
 
     attr_reader :path
+    
+    attr_reader :struct
+    
+    def self.__allocate__
+      instance = super
+      instance.instance_variable_set :@struct, StatStruct.new
```

Comment:
Shouldn't this be something like `FFI::MemoryPointer.new(StatStruct, 1)`?

Reply:
Or, would it be `StatStruct.new FFI::MemoryPointer.new(StatStruct, 1)`?

Reply:
Am I using struct incorrectly here, @eregon?

Reply:
That should be fine, https://github.com/graalvm/truffleruby/blob/86bbd8c880a106b3df51fdc95f50cdc7d89c4ad1/src/main/ruby/core/truffle/ffi/ffi_struct.rb#L241 does allocate a pointer of the size of the struct.